core/Db/TransactionLevel.php

```diff
@@ -38,7 +38,15 @@ public function canLikelySetTransactionLevel()
         return strtolower($dbSettings->getEngine()) === 'innodb';
     }
 
+    /**
+     * @deprecated Use `setTransactionLevelForNonLockingReads`
+     */
     public function setUncommitted()
+    {
+        return $this->setTransactionLevelForNonLockingReads();
+    }
+
+    public function setTransactionLevelForNonLockingReads(): bool
     {
         if ($this->db->supportsUncommitted === false) {
```

Comment:
Guess it's discussable if the public db property should still be called `supportsUncommitted`, when it actually no longer holds that information.

Reply:
I agree it's name wrong, on one side as well I'm not exactly sure why it's exposed as public in the first place. 

Would you suggest making it private and make this class less leaky, along with a proper deprecation message?

Reply:
I guess it's exposed public and set for the DB class, so the check is only performed once per DB connection. As the property shouldn't be used by any plugin I would say we can simply rename it, or make it private and introduce methods to handle that correctly.
We could even consider to handle everything through DB class, instead of using the TransactionLevel class directly.